lib/spack/spack/util/module_cmd.py

```diff
@@ -40,13 +48,19 @@ def module(*args):
 
         # Cray modules spit out warnings that we cannot supress.
         # This hack skips to the last output (the environment)
-        env_output = str(module_p.communicate()[0].decode())
-        env = env_output.strip().split('\n')[-1]
+        env_out = str(module_p.communicate()[0].decode()).strip().split('\n')
+        env = env_out[-2]
+        new_ld_library_path = env_out[-1]
```

Comment:
The cognitive load for reading and understanding `module()` is getting pretty high.

I'd like to separate the concerns of:
1. examining what `module` would do, and
2. dealing with the complexities of preserving the library path of `sys.executable`.

Curious if we could simplify this by implementing the `Popen` stuff above by subclassing `executable.Executable`.  In particular, `Executable` already has facilities for injecting a custom environment, and it already does the `Popen`.  It may not be worth it as I am not sure we have a way to elegantly handle the final `LD_LIBRARY_PATH="$SPACK_LD_LIBRARY_PATH` before the python execution.  At the very least, I think there is a separate function to be made for the `sys.executable` part.

Reply:
As per our discussion, I don't think this can be a separate function without making it worse. The tight coupling of the module/shell integration with the python workarounds suggests this is one-off code.